batch-inference/batch-inference-transcript-summarization.ipynb

```diff
@@ -0,0 +1,516 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f708bbb3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Write model inputs to a jsonl file\n",
+    "filename = 'batch-' + str(int(datetime.now().timestamp())) + '.jsonl'\n",
+    "write_jsonl(model_input_jsonl, f'{filename}')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "95a90686",
+   "metadata": {},
+   "source": [
+    "### Uploading to Amazon S3\n",
+    "\n",
+    "The `upload_to_s3` function uploads a file or directory to an AWS S3 bucket. It takes three arguments:\n",
+    "\n",
+    "1. `path`: The path to the file or directory to be uploaded.\n",
+    "2. `bucket_name`: The name of the S3 bucket.\n",
+    "3. `bucket_subfolder` (optional): The name of the subfolder within the S3 bucket where the prepared data should be uploaded."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8517a4b3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def upload_to_s3(path, bucket_name, bucket_subfolder=None):\n",
+    "    # check if the path is a file\n",
+    "    if os.path.isfile(path):\n",
+    "        # If the path is a file, upload it directly\n",
+    "        object_name = os.path.basename(path) if bucket_subfolder is None else f\"{bucket_subfolder}/{os.path.basename(path)}\"\n",
+    "        try:\n",
+    "            s3.upload_file(path, bucket_name, object_name)\n",
+    "            print(f\"Successfully uploaded {path} to {bucket_name}/{object_name}\")\n",
+    "            return True\n",
+    "        except Exception as e:\n",
+    "            print(f\"Error uploading {path} to S3: {e}\")\n",
+    "            return False\n",
+    "    elif os.path.isdir(path):\n",
+    "        # If the path is a directory, recursively upload all files within it\n",
+    "        for root, dirs, files in os.walk(path):\n",
+    "            for file in files:\n",
+    "                file_path = os.path.join(root, file)\n",
+    "                relative_path = os.path.relpath(file_path, path)\n",
+    "                object_name = relative_path if bucket_subfolder is None else f\"{bucket_subfolder}/{relative_path}\"\n",
+    "                try:\n",
+    "                    s3.upload_file(file_path, bucket_name, object_name)\n",
+    "                    print(f\"Successfully uploaded {file_path} to {bucket_name}/{object_name}\")\n",
+    "                except Exception as e:\n",
+    "                    print(f\"Error uploading {file_path} to S3: {e}\")\n",
+    "        return None\n",
+    "    else:\n",
+    "        print(f\"{path} is not a file or directory.\")\n",
+    "        return None"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a4b688b9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# uploads the data from local to S3 bucket for batch inference\n",
+    "upload_to_s3(path=f\"/home/ec2-user/SageMaker/{filename}\", \n",
```

Comment:
When I run this command in my notebook instance, I got this message “/home/ec2-user/SageMaker/batch-1724422794.jsonl is not a file or directory.”. Since we process everything in local, could it just be "path=filename" here?